app/models/grda_warehouse/hud/client.rb

```diff
@@ -1060,7 +1060,9 @@ def self.release_duration
       @release_duration = GrdaWarehouse::Config.get(:release_duration)
     end
 
-    def release_valid?
+    def release_valid?(coc_codes: nil)
+      return active_confirmed_consent_in_cocs(coc_codes).exists? if coc_codes.present?
```

Comment:
Pretty sure you want to limit to the client in question, and `active_confirmed_consent_in_cocs` is a scope, so I _think_ you have to call it on the class.
```suggestion
      return self.class.where(id: id).active_confirmed_consent_in_cocs(coc_codes).exists? if coc_codes.present?
```